azext_iot/operations/hub.py

```diff
@@ -340,6 +328,66 @@ def iot_device_delete(
         raise CLIError(err)
 
 
+def iot_device_key_renew(cmd, hub_name, device_id, regenerate_key, resource_group_name=None, login=None):
+    discovery = IotHubDiscovery(cmd)
+    target = discovery.get_target(
+        hub_name=hub_name, resource_group_name=resource_group_name, login=login
+    )
+    device = _iot_device_show(target, device_id)
+    if (device["authentication"]["type"] == "sas"):
+        pk = device["authentication"]["symmetricKey"]["primaryKey"]
+        sk = device["authentication"]["symmetricKey"]["secondaryKey"]
+        if regenerate_key == RenewKeyType.Primary.value:
+            pk = _generateKey()
+        if regenerate_key == RenewKeyType.Secondary.value:
+            sk = _generateKey()
+        if regenerate_key == RenewKeyType.Swap.value:
+            temp = pk
+            pk = sk
+            sk = temp
+
+        updated_device = _assemble_device(
+            device["deviceId"],
+            device["authentication"]["type"],
+            device["capabilities"]["iotEdge"],
+            pk,
+            sk,
+            device["status"].lower(),
+            device["statusReason"],
+            device.get("deviceScope", None)
+        )
+        updated_device.etag = device["etag"]
+        return _iot_device_update(target, device_id, updated_device)
+    else:
```

Comment:
nit, but I'd prefer to reverse the if statement here, as in have 

```python
...
device = _iot_device_show(target, device_id)
if device["authentication"]["type"] != "sas":
    raise CLIError("Device authentication should be of type sas")

pk = device["authentication"]["symmetricKey"]["primaryKey"]
sk = device["authentication"]["symmetricKey"]["secondaryKey"]
if regenerate_key == RenewKeyType.Primary.value:
   ...
...
```

Reply:
Updated